codalab/worker/docker_image_manager.py

```diff
@@ -218,12 +224,34 @@ def download():
                             'message'
                         ] = "Can't download image: {}".format(ex)
 
-                self._downloading.add_if_new(image_spec, threading.Thread(target=download, args=[]))
+                # Check docker image size before pulling from docker hub.
+                # Images with size larger than '10g' will be rejected to download.
+                image_size_bytes = docker_utils.get_image_size_without_pulling(image_spec)
+                if 0 < image_size_bytes < self._max_image_size:
```

Comment:
Why is `0 < ` necessary?  Should be always positive no?

Reply:
There is a chance that we don't get the requested image from the docker registry, then the return image size will be empty. In this particular case, I was defining it as `0`. I guess `None` will be better.